src/imgui/ImGuiSymbols.cc

```diff
@@ -158,11 +204,32 @@ static void drawTable(ImGuiManager& manager, const SymbolManager& symbolManager,
 
 			if (ImGui::TableNextColumn()) { // name
 				im::ScopedFont sf(manager.fontMono);
-				ImGui::TextUnformatted(sym.name(symbolManager));
+				im::StyleColor(sym.segment(symbolManager) && *sym.segment(symbolManager) > 0, ImGuiCol_Text, getColor(imColor::YELLOW), [&]{
```

Comment:
You're calling `sym.segment(symbolManager)` twice. That's not super expensive, but also not needed. And very easy to avoid: do it once and store the result in a local variable.

Though in this specific case there's a useful shortcut: you can directly compare a `std::optional<uint16_t>` with `> 0` (a `std::nullopt`-value compares smaller than any other `uint16_t` value).

But I also wonder why it's useful to show non-zero segments in a different color. What's special about segment zero. Especially after you try harder to detect symbol files with no segment information (and put those to `std::nullopt`).